### ISSUE TITLE
custom sound only plays when app is in foreground (ios8.1.2)

### ISSUE THREAD

**Issue by ghost** [phonegap-build/PushPlugin#437]
Is the custom sound supposed to also play when the app is not running? When not running I just get the default push notification sound. I'm using push plugin 2.4.0. The custom sound works fine when the app is in the foreground.

```
// handle APNS notifications for iOS
function onNotificationAPN(e) {
if (e.alert) {
    $("#app-status-ul").append('<li>push-notification: ' + e.alert + '</li>');
    // showing an alert also requires the org.apache.cordova.dialogs plugin
    try {
        navigator.notification.alert(e.alert);
    } catch (exception) {
        $("#app-status-ul").append('<li>failed</li>');

    }
}

$("#app-status-ul").append('<li>push-sound: ' + e.sound + '</li>');
if (e.sound) {
    // playing a sound also requires the org.apache.cordova.media plugin
    var snd = new Media(e.sound);
    snd.play();
}

if (e.badge) {
    pushNotification.setApplicationIconBadgeNumber(successHandler, e.badge);
}
}
```

**Comment by tmakaro** [comment]
I am too new at IOS development to be helpful at this point unfortunately. :(

**Comment by tmakaro** [comment]
Has this been looked into yet?

**Comment by andreszs** [comment]
No, I don't think it will ever be. The pushplugin has no control of notifications events simply because it's not even running when the app is unloaded, therefore the phone will play the default sound and default vibration unless the app is in foreground.

**Comment by tmakaro** [comment]
Then why does it work that way on a native app? I wrote an empty native app that all it did was register for push notifications and include a sound file. It plays that sound file regardless of if the program is in the foreground or the background.

**Comment by andreszs** [comment]
Because when sent to the background, native apps keep running and listening to your events, while Cordova apps are paused.

**Comment by tmakaro** [comment]
Darn :(

**Comment by kerloom** [comment]
You can modify the native code (at least for android) and change the notification sound, vibration, flags, etc.

**Comment by Mikechaos** [comment]
So is it completely impossible to play a custom sound, included in the payload of the push notification, if using phonegap build? Or even, if not native at all?
Of what I had understood, it is APNS and iOS that take care of playing the custom sound on push notification (APNS sends the payload to the device, which then look into the app's asset folder and plays the given sound).

**Comment by cinder92** [comment]
same issue here! +1
